src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java

```diff
@@ -548,20 +576,48 @@ private void initDockerFileFile(Logger log) {
 
     private File findDockerFileFile(Logger log) {
         if (dockerFile != null) {
+            if (EnvUtil.isWindows() && !EnvUtil.isValidWindowsFileName(dockerFile)) {
+                throw new IllegalArgumentException(String.format("Invalid Windows file name %s for <dockerFile>", dockerFile));
+            }
+
             File dFile = new File(dockerFile);
-            if (dockerFileDir == null) {
+            if (dockerFileDir == null && contextDir == null) {
                 return dFile;
             } else {
-                if (dFile.isAbsolute()) {
-                    throw new IllegalArgumentException("<dockerFile> can not be absolute path if <dockerFileDir> also set.");
+                if(contextDir != null) {
+                    if(dockerFileDir != null) {
+                        log.warn("using contextDir as it has more precedence for Dockerfile");
```

Comment:
Precedence over what ? Suggestion: "Both contextDir (%s) and deprecated dockerFileDir (%s) are configured. Using contextDir." (with inserting the concrete values)